File: cos20007/11.1P/clock.py
class Counter:
    def __init__(self, name):
        self.__count = 0
        self.name = name
    
    def increment(self):
        self.__count += 1

    def reset(self):
        self.__count = 0

    def ticks(self):
        return self.__count

class Clock:
    def __init__(self):
        self.__hours = Counter("hours")
        self.__minutes = Counter("minutes")
        self.__seconds = Counter("seconds")
    
    def reset(self):
        self.__hours.reset()
        self.__minutes.reset()
        self.__seconds.reset()

    def tick(self):
        self.__seconds.increment()

        if self.__seconds.ticks() == 60:
            self.__seconds.reset()
            self.__minutes.increment()
        
        if self.__minutes.ticks() == 60:
            self.__minutes.reset()
            self.__hours.increment()
        
        if self.__hours.ticks() == 24:
            self.__hours.reset()
        
    def time(self):
        return "{:02d}".format(self.__hours.ticks()) + ":" + "{:02d}".format(self.__minutes.ticks()) + ":" + "{:02d}".format(self.__seconds.ticks())

def main():
    clock = Clock()

    for i in range(0, 1243):
        clock.tick()

    print(clock.time())

main()

File: cos20007/3.1P/program/Clock.cs
namespace ClockClass
{
    public class Clock
    {
        private Counter _hours;
        private Counter _minutes;
        private Counter _seconds;

        public Clock()
        {
            _hours = new Counter("hours");
            _minutes = new Counter("minutes");
            _seconds = new Counter("seconds");
        }

        public void Reset()
        {
            _hours.Reset();
            _minutes.Reset();
            _seconds.Reset();
        }

        public void Tick()
        {
            _seconds.Increment();

            if (_seconds.Ticks == 60)
            {
                _seconds.Reset();
                _minutes.Increment();
            }

            if (_minutes.Ticks == 60)
            {
                _minutes.Reset();
                _hours.Increment();
            }

            if (_hours.Ticks == 24)
            {
                _hours.Reset();
            }
        }

        public string Time
        {
            get
            {
                string hours = _hours.Ticks.ToString().PadLeft(2, '0');
                string minutes = _minutes.Ticks.ToString().PadLeft(2, '0');
                string seconds = _seconds.Ticks.ToString().PadLeft(2, '0');
                return hours + ":" + minutes + ":" + seconds;
            }
        }
    }
}
